File: python/diccionarios/actualizar_contacto.py
agenda = {
    "Juan": {"telefono": "3001234567", "correo": "juan@gmail.com"},
    "Maria": {"telefono": "3019876543", "correo": "maria@gmail.com"}
}
nombre = input("Ingresa el nombre del contacto que deseas actualizar: ")
if nombre in agenda:
    nuevo_telefono = input("Ingresa el nuevo teléfono: ")
    nuevo_correo = input("Ingresa el nuevo correo: ")
    agenda[nombre]["telefono"] = nuevo_telefono
    agenda[nombre]["correo"] = nuevo_correo
    print("Contacto actualizado correctamente.")
else:
    print("El contacto no existe en la agenda.")
print("\nAgenda actualizada:")
print(agenda)

File: visual_basic/colecciones_complejas/actualizar_contacto/actualizar_contacto.vb
Imports System
Imports System.Collections.Generic
Module actualizar_contacto
    Class Contacto
        Public Nombre As String
        Public Telefono As String
        Public Correo As String
    End Class
    Sub Main(args As String())
        Dim agenda As New List(Of Contacto)
        agenda.Add(New Contacto With {.Nombre = "Juan", .Telefono = "123", .Correo = "juan@mail.com"})
        agenda.Add(New Contacto With {.Nombre = "Maria", .Telefono = "456", .Correo = "maria@mail.com"})
        Console.WriteLine("📒 CONTACTOS ACTUALES:")
        MostrarContactos(agenda)
        Console.WriteLine()
        Console.Write("Ingrese el nombre del contacto a actualizar: ")
        Dim nombreBuscar As String = Console.ReadLine()
        Dim encontrado As Boolean = False
        For Each contacto In agenda
            If contacto.Nombre.ToLower() = nombreBuscar.ToLower() Then
                encontrado = True
                Console.WriteLine("✅ Contacto encontrado. Ingrese nuevos datos:")
                Console.Write("Nuevo teléfono: ")
                contacto.Telefono = Console.ReadLine()
                Console.Write("Nuevo correo: ")
                contacto.Correo = Console.ReadLine()
                Console.WriteLine("✨ Contacto actualizado correctamente.")
                Exit For
            End If
        Next
        If Not encontrado Then
            Console.WriteLine("❌ Contacto no encontrado.")
        End If
        Console.WriteLine()
        Console.WriteLine("📒 CONTACTOS ACTUALIZADOS:")
        MostrarContactos(agenda)
        Console.ReadKey()
    End Sub
    Sub MostrarContactos(lista As List(Of Contacto))
        For Each contacto In lista
            Console.WriteLine("------------------------")
            Console.WriteLine("Nombre: " & contacto.Nombre)
            Console.WriteLine("Teléfono: " & contacto.Telefono)
            Console.WriteLine("Correo: " & contacto.Correo)
        Next
    End Sub
End Module